plugins/modules/df_deployment.py

```diff
@@ -0,0 +1,467 @@
+    type: int
+    required: False
+    default: 15
+    aliases:
+      - polling_delay
+  timeout:
+    description:
+      - The internal polling timeout (in seconds) while the module waits for the Dataflow Service to achieve the 
+        declared state.
+    type: int
+    required: False
+    default: 3600
+    aliases:
+      - polling_timeout
+notes:
+  - This feature this module is for is in Technical Preview
+extends_documentation_fragment:
+  - cloudera.cloud.cdp_sdk_options
+  - cloudera.cloud.cdp_auth_options
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details.
+
+# Deploy a Dataflow with defaults
+- cloudera.cloud.df_deployment:
+    name: my-flow
+
+# Remove a Dataflow Service with Async wait
+- cloudera.cloud.df_deployment:
+    name: my-flow-name
+    df_name: my-env-name
+    state: absent
+    wait: yes
+  async: 3600
+  poll: 0
+  register: __my_teardown_request
+
+'''
+
+RETURN = r'''
+---
+deployments:
+  description: The information about the named DataFlow Deployment or DataFlow Deployments
+  type: list
+  returned: always
+  elements: complex
+  contains:
+    crn:
+      description:  The DataFlow Service's parent environment CRN.
+      returned: always
+      type: str
+    name:
+      description: The DataFlow Service's parent environment name.
+      returned: always
+      type: str
+    cloudPlatform:
+      description: The cloud platform of the environment.
+      returned: always
+      type: str
+    region:
+      description: The region of the environment.
+      returned: always
+      type: str
+    deploymentCount:
+      description: The deployment count.
+      returned: always
+      type: str
+    minK8sNodeCount:
+      description: The  minimum number of Kubernetes nodes that need to be provisioned in the environment.
+      returned: always
+      type: int
+    maxK8sNodeCount:
+      description:  The maximum number of kubernetes nodes that environment may scale up under high-demand situations.
+      returned: always
+      type: str
+    status:
+      description: The status of a DataFlow enabled environment.
+      returned: always
+      type: dict
+      contains:
+        state:
+          description: The state of the environment.
+          returned: always
+          type: str
+        message:
+          description: A status message for the environment.
+          returned: always
+          type: str
+    k8sNodeCount:
+      description: The number of kubernetes nodes currently in use by DataFlow for this environment.
+      returned: always
+      type: int
+    instanceType:
+      description: The instance type of the kubernetes nodes currently in use by DataFlow for this environment.
+      returned: always
+      type: str
+    dfLocalUrl:
+      description: The URL of the environment local DataFlow application.
+      returned: always
+      type: str
+    authorizedIpRanges:
+      description: The authorized IP Ranges.
```

Comment:
Are these CIDRs?

Reply:
That whole documentation section was wrong, now updated